frame/staking/src/pallet/mod.rs

```diff
@@ -850,31 +847,31 @@ pub mod pallet {
 
 			frame_system::Pallet::<T>::inc_consumers(&stash).map_err(|_| Error::<T>::BadState)?;
 
-			// You're auto-bonded forever, here. We might improve this by only bonding when
-			// you actually validate/nominate and remove once you unbond __everything__.
-			<Bonded<T>>::insert(&stash, &stash);
-			<Payee<T>>::insert(&stash, payee);
-
 			let current_era = CurrentEra::<T>::get().unwrap_or(0);
 			let history_depth = T::HistoryDepth::get();
 			let last_reward_era = current_era.saturating_sub(history_depth);
 
 			let stash_balance = T::Currency::free_balance(&stash);
 			let value = value.min(stash_balance);
 			Self::deposit_event(Event::<T>::Bonded { stash: stash.clone(), amount: value });
-			let item = StakingLedger {
-				stash: stash.clone(),
-				total: value,
-				active: value,
-				unlocking: Default::default(),
-				claimed_rewards: (last_reward_era..current_era)
+			let ledger: StakingLedger<T> = StakingLedger::<T>::new(
+				stash.clone(),
+				value,
+				value,
+				Default::default(),
+				(last_reward_era..current_era)
 					.try_collect()
 					// Since last_reward_era is calculated as `current_era -
 					// HistoryDepth`, following bound is always expected to be
 					// satisfied.
 					.defensive_map_err(|_| Error::<T>::BoundNotMet)?,
-			};
-			Self::update_ledger(&controller_to_be_deprecated, &item);
+			);
+
+			// You're auto-bonded forever, here. We might improve this by only bonding when
+			// you actually validate/nominate and remove once you unbond __everything__.
+			ledger.bond()?;
```

Comment:
At the beginning of this function, we made sure that this account is not staked already. Therefore, this shoudl be defensive. 

If this is the only reason why `fn bond` exists thourgh, I would suggest just using `fn update`. 


```suggestion
			ledger.update();
```

Reply:
With the new version where `Payee` mutations are encapsulated in the staking ledger, I think that `ledger.bond(reward_destination)` makes more sense. So that we don't need to pass an option with the reward destination every time we want to update the ledger or make the API more convoluted. wdyt? (will update the PR accordingly)